src/stepFunctions/commands/visualizeStateMachine.ts

```diff
@@ -29,257 +29,341 @@ export interface messageObject {
-     * Ensure tests are written for this use case as well.
-     */
-    const activeTextEditor: vscode.TextEditor | undefined = vscode.window.activeTextEditor
-
-    let documentUri: vscode.Uri
-    let textDocument: vscode.TextDocument
-
-    if (activeTextEditor) {
-        documentUri = activeTextEditor.document.uri
-        textDocument = activeTextEditor.document
-    } else {
-        logger.error('Could not get active text editor for state machine render.')
-        throw new Error('Could not get active text editor for state machine render.')
-    }
+            return existingVisualization.getPanel()
+        }
 
-    try {
-        await cache.updateCache(globalStorage)
+        // Existing visualization does not exist, construct new visualization
+        try {
+            await cache.updateCache(globalStorage)
 
-        return setupWebviewPanel(documentUri, textDocument)
-    } catch (err) {
-        vscode.window.showInformationMessage(
-            localize(
-                'AWS.stepfunctions.visualisation.errors.rendering',
-                'There was an error rendering State Machine Graph, check logs for details.'
+            return this.createNewVisualization(textDocument)
+        } catch (err) {
+            vscode.window.showInformationMessage(
+                localize(
+                    'AWS.stepfunctions.visualisation.errors.rendering',
+                    'There was an error rendering State Machine Graph, check logs for details.'
+                )
             )
-        )
 
-        logger.debug('Unable to setup webview panel.')
-        logger.error(err as Error)
+            logger.debug('Unable to setup webview panel.')
+            logger.error(err as Error)
+        }
+
+        return
     }
 
-    return
-}
+    private createNewVisualization(textDocument: vscode.TextDocument): vscode.WebviewPanel | void {
+        const newVisualization = new AslVisualization(textDocument)
+        this.managedVisualizations.add(newVisualization)
 
-async function setupWebviewPanel(
-    documentUri: vscode.Uri,
-    textDocument: vscode.TextDocument
-): Promise<vscode.WebviewPanel> {
-    const logger: Logger = getLogger()
-    let isPanelDisposed = false
-
-    // Create and show panel
-    const panel = vscode.window.createWebviewPanel(
-        'stateMachineVisualization',
-        makeWebviewTitle(documentUri),
-        {
-            preserveFocus: true,
-            viewColumn: vscode.ViewColumn.Beside,
-        },
-        {
-            enableScripts: true,
-            localResourceRoots: [
-                ext.visualizationResourcePaths.localWebviewScriptsPath,
-                ext.visualizationResourcePaths.visualizationLibraryCachePath,
-                ext.visualizationResourcePaths.stateMachineCustomThemePath,
-            ],
-            retainContextWhenHidden: true,
-        }
-    )
+        newVisualization.onVisualizationDispose(() => {
```

Comment:
This returns a disposable that should be handled somewhere (possibly ext.subscriptions)

Reply:
Given that the VisualizationManager is initialized in the plugin activation and does not get disposed while vscode is open, does this disposable still need to be handled? If so is there a way to dispose of it on application close?